common/text/syntax_tree_context.h

```diff
@@ -31,38 +33,16 @@ namespace verible {
 // Note: Public methods are named to follow STL convention for std::stack.
 // TODO(fangism): implement a general ForwardMatcher and ReverseMatcher
 // interface that can express AND/OR/NOT.
-class SyntaxTreeContext {
+class SyntaxTreeContext : public AutoPopStack<const SyntaxTreeNode*> {
```

Comment:
Document that one of the invariants is that elements of the (internal) stack are restricted to non-nullptrs.  (This was the case before refactoring.)

Reply:
Done